pkg/controller/subjectpermission/subjectpermission_controller.go

```diff
@@ -255,22 +283,24 @@ func newClusterRoleBinding(clusterRoleName, subjectName string, subjectKind stri
 // populateCrClusterRoleNames to see if ClusterRoleName exists as a ClusterRole
 // returns list of ClusterRoleNames that do not exist
 func populateCrClusterRoleNames(subjectPermission *managedv1alpha1.SubjectPermission, clusterRoleList *v1.ClusterRoleList) []string {
-	// we get clusterRoleName by managedv1alpha1.ClusterPermission{}
 	crClusterRoleNames := subjectPermission.Spec.ClusterPermissions
 
 	// items is list of clusterRole on k8s
 	onClusterItems := clusterRoleList.Items
 
 	var crClusterRoleNameList []string
+	var found bool
 
-	// for every cluster role names on cluster, loop through all crClusterRoleNames, if it doesn't exist then append
-	for _, i := range onClusterItems {
-		//name := i.Name
-		for _, a := range crClusterRoleNames {
-			if i.Name != a {
-				crClusterRoleNameList = append(crClusterRoleNameList, a)
+	// for every CR clusterRoleNames, loop through all k8s lusterRoles, if it doesn't exist then append
+	for _, i := range crClusterRoleNames {
```

Comment:
Could this be a function?

Reply:
the nested for loop is the whole logic/reason of the function, not sure if moving it to another helper function makes sense

Reply:
Isn't this the same logic as https://github.com/openshift/rbac-permissions-operator/pull/32/files#diff-ef4d9a930cf76db3ac6ca4d543b6267dR25-R35

Reply:
It's very similar but not the same @jharrington22

Reply:
this one loops through []string, the other one loops through []Permissions

Reply:
future feature: make this a type agnostic way to do the nested for loop